README.md

```diff
@@ -1,16 +1,192 @@
-# Title
+# Arrow Function Shorthand
+
+## Introduction
+
+The original style for declaring functions in JavaScript is the _function
+declaration_.
+
+```js
+function foo() {
+  return 'bar';
+}
+```
+
+But JavaScript has two other ways to write functions: the _function expression_
+and the _arrow function_. None of these is more correct or better than others.
 
 ## Learning Goals
 
--SWBAT 1
--SWBAT 2
+- Declare a function using a function expression
+- Declare a function using an arrow function
+- Describe situations where arrow functions are used
 
-## Introduction
+## Declare a Function Using a Function Expression
+
+Thus far we've only ever used a _function declaration_:
+
+```js
+function foo() {
+  return 'bar';
+}
+```
+
+A function can also be written:
+
+```js
+let foo = function() {
+  return 'bar';
+}
+```
+
+The `function() {...}` to the right of the assignment operator (`=`) is called
+a _function expression_. The best way to understand what a function expression
+is is by analogy.
+
+```js
+let sum = 1 + 1
+```
+
+Evaluate the expression `1 + 1`, returning `2` and assign it to the variable `sum`
+
+```js
+let difference = 10 - 1;
+```
+
+Evaluate the expression `10 - 1`, returning `9` and assign it to the variable
+`difference`.
+
+```js
+let foo = function() {
+  return 'bar';
+}
+```
+
+Evaluate the expression `function() { return 'bar' }`, returning a thing that
+can be called and assign it to the variable `foo`. The function expression  
+(again, the thing to the right of `=`) is known as "an anonymous
+function." It doesn't have a name associated with it like you see in a
+_function declaration_.
+
+However, when we assign an anonymous function to a name (that is, a variable),
+we have name that points to a callable thing. We could call this anonymous
+function by invoking `foo()`. That anonymous function is now, for all
+reasonable purposes, named `foo`.
+
+There are a few subtle differences between _function declarations_ and
+_function expressions_, but they are very minute. Neither is really better than
+the other. JavaScript supports variety. Neither is better or more preferred
+than the other.
+
+Try thinking about the following code. Will this work? Does it work? Try to
+explain to yourself what's happening using the words "anonymous function" and
+"`call`." If you need help, see [Immediately Invoked Function Expression][IIFE]
+in the Resources section below.
+
+```js
+(function() { console.log("Hello world") })()
+```
+
+## Declare a Function Using An Arrow Function
 
-## SWBAT 1
+Around 2015, developers got tired of typing `f-u-n-c-t-i-o-n` over and over.
+They lobbied the ECMAScript body for a way to write functions in a very short
+way. Here is the result, the "Arrow Function."
+
+```js
+let add = (parameter1, parameter2) => parameter1 + parameter2
+add(2,3) //=> 5
+```
+
+It seems like it wouldn't work, but it does. This builds on the syntax of the _function expression_
```

Comment:
I don't think "It seems like it wouldn't work, but it does." needs to be here. It's not necessary to know that it seems like it shouldn't work and if it's not evident to a student that it seems like it shouldn't work, it could be confusing.